src/shell/Shell.cpp

```diff
@@ -571,7 +575,14 @@ static void executeInvokeAction(wabt::InvokeAction* action, Walrus::Function* fn
             RELEASE_ASSERT(data->fn->functionType()->result().size() == data->expectedResult.size());
             // compare result
             for (size_t i = 0; i < result.size(); i++) {
-                RELEASE_ASSERT(equals(result[i], data->expectedResult[i]));
+                if (!equals(result[i], data->expectedResult[i])) {
+                    printf("invoke %s(", data->action->name.data());
+                    printConstVector(data->action->args);
+                    printf(") expect value(");
+                    printConstVector(data->expectedResult);
+                    printf(") (line: %d) : FAILED\n", data->action->loc.line);
+                    abort();
```

Comment:
I think printing `i`, `result[i]`, and `data->expectedResult[i]` are the key thing here

Reply:
At this point, `printf` is used to print messages but other case, `fprintf` is used instead.
For the code consistency, please use just one function for printing messages.
One more, we prefer `RELEASE_ASSERT`  or `ASSERT` to `abort` function.
Please change `abort` function too.

Reply:
Okay.
Thank you!